addons/web/static/tests/actions/helpers.js

```diff
@@ -114,9 +118,18 @@ function addLegacyMockEnvironment(comp, testConfig, legacyParams = {}) {
     },
     legacyParams.dataManager
   );
+
   const legacyEnv = legacy.makeTestEnvironment({ dataManager, bus: legacy.core.bus });
+
+  if (legacyParams.serviceRegistry) {
+    const legacyServiceRegistry = legacy.core.serviceRegistry;
+    legacy.core.serviceRegistry = legacyParams.serviceRegistry;
+    legacy.AbstractService.prototype.deployServices(legacyEnv);
+    legacy.core.serviceRegistry = legacyServiceRegistry;
```

Comment:
this last instruction could (should?) be done in registerCleanup (to be executed when the test is over)